docs/data/ingest-library/developer_guide/architecture.mdx

```diff
@@ -0,0 +1,14 @@
+---
+title: Architecture
+sidebar_position: 0
+---
+
+![Insert architecture diagram]
+
+## Package Structure
+
+The ingest library is broadly split into three main packages:
+
+1. **Ledger Backends**: A wrapper around sources of ledger data to fetch ledgers from the Stellar network.
+2. **Ledger Readers**: Utility classes that help iterate through individual changes or transactions in a ledger.
```

Comment:
since the library is written in Go we should avoid the term class because although Go supports object oriented programming it doesn't refer to objects as classes.

```suggestion
2. **Ledger Readers**: Iterators to extract individual changes or transactions from a ledger.
```